kernel/src/hil/symmetric_encryption.rs

```diff
@@ -0,0 +1,16 @@
+use returncode::ReturnCode;
+
+pub trait SymmetricEncryptionDriver {
```

Comment:
Need comments for all of these functions explaining what they do, what their inputs are/assume and how they related to each other. The comments should be explicit enough to help a client use the interface as well as an implementor implement it.